js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,327 @@
+// Package fs provides a k6 module that allows users to interact with files from the
+// local filesystem.
+package fs
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/js/promises"
+	"go.k6.io/k6/lib/fsext"
+)
+
+type (
+	// RootModule is the global module instance that will create instances of our
+	// module for each VU.
+	RootModule struct {
+		cache *cache
+	}
+
+	// ModuleInstance represents an instance of the fs module for a single VU.
+	ModuleInstance struct {
+		vu    modules.VU
+		cache *cache
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// New returns a pointer to a new [RootModule] instance.
+func New() *RootModule {
+	return &RootModule{
+		cache: &cache{},
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface and returns a new
+// instance of our module for the given VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &ModuleInstance{vu: vu, cache: rm.cache}
+}
+
+// Exports implements the modules.Module interface and returns the exports of
+// our module.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]any{
+			"open": mi.Open,
+			"SeekMode": map[string]any{
+				"Start":   SeekModeStart,
+				"Current": SeekModeCurrent,
+				"End":     SeekModeEnd,
+			},
+		},
+	}
+}
+
+// Open opens a file and returns a promise that will resolve to a [File] instance
+func (mi *ModuleInstance) Open(path goja.Value) *goja.Promise {
+	promise, resolve, reject := promises.New(mi.vu)
+
+	// Files can only be opened in the init context.
+	if mi.vu.State() != nil {
+		reject(newFsError(ForbiddenError, "open() failed; reason: opening a file in the VU context is forbidden"))
+		return promise
+	}
```

Comment:
```suggestion
	if mi.vu.State() != nil {
		reject(newFsError(ForbiddenError, "open() failed; reason: opening a file is allowed only in the Init context"))
		return promise
	}
```
It sounds the error message is not clear enough if we need to add a dedicated comment for it. So I would like to merge it in the error message so it is clear for the code-redear but also for any eventual user that may hit it.
In any case, I expect if the user hits this probably they aren't expert user so they don't know what a Init or VU context is, so if we can find an alternative without mentioning them then it should be better. Unfortunately, I don't have a concrete proposal.